api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCalculation.java

```diff
@@ -1,17 +1,39 @@
 package uk.gov.dhsc.htbhf.claimant.service.payments;
 
 import lombok.Builder;
-import lombok.Data;
+import lombok.Value;
 import uk.gov.dhsc.htbhf.claimant.entity.PaymentCycleStatus;
 
+import java.time.LocalDateTime;
+
+import static uk.gov.dhsc.htbhf.claimant.entity.PaymentCycleStatus.BALANCE_TOO_HIGH_FOR_PAYMENT;
+
 /**
  * The result of a payment calculation from {@link PaymentCalculator}.
  */
-@Data
+@Value
 @Builder
 public class PaymentCalculation {
 
     private PaymentCycleStatus paymentCycleStatus;
-
     private int paymentAmount;
+    private int availableBalanceInPence;
+    private LocalDateTime balanceTimestamp;
+
+    public static PaymentCalculation aFullPaymentCalculation(int paymentAmount) {
+        return PaymentCalculation.builder()
+                .paymentAmount(paymentAmount)
+                .paymentCycleStatus(PaymentCycleStatus.FULL_PAYMENT_MADE)
+                .balanceTimestamp(LocalDateTime.now())
```

Comment:
Shouldn't we also be storing the available balance on this object too? We should have it when they have a small balance and a full payment is still being made (not just the first payment).

Reply:
The balance isn't being set explicitly (and maybe it should be), but it will be 0, which is the correct amount. The confusion may be my fault, as I asked for the method to be renamed from `aFirstPayment`.

Reply:
Renamed the method to aFullPaymentWithZeroBalance.

Reply:
Is it worth an explicit `.availableBalanceInPence(0)` as you've done with the `paymentAmount` below?

Reply:
Done.